### DISCUSSION TITLE
mmdetection3d, mono_det_demo.py shows buggy result

### DISCUSSION THREAD

**Discussion by damavand1** [open-mmlab/mmdetection3d#2943] [General]
Hi
Based in this document
https://mmdetection3d.readthedocs.io/en/latest/user_guides/inference.html#monocular-3d-demo

I run this part of code

python demo/mono_det_demo.py demo/data/kitti/000008.png demo/data/kitti/000008.pkl  configs/pgd/pgd_r101-caffe_fpn_head-gn_4xb3-4x_kitti-mono3d.py pgd_r101_caffe_fpn_gn-head_3x4_4x_kitti-mono3d_20211022_102608-8a97533b.pth  --show --cam-type CAM2

I get this result 
`(openmmlab) user1@pc1:~/Desktop/a/mmdetection3d$ python demo/mono_det_demo.py demo/data/kitti/000008.png demo/data/kitti/000008.pkl  configs/pgd/pgd_r101-caffe_fpn_head-gn_4xb3-4x_kitti-mono3d.py pgd_r101_caffe_fpn_gn-head_3x4_4x_kitti-mono3d_20211022_102608-8a97533b.pth  --show --cam-type CAM2
Loads checkpoint by local backend from path: pgd_r101_caffe_fpn_gn-head_3x4_4x_kitti-mono3d_20211022_102608-8a97533b.pth
04/01 12:05:41 - mmengine - INFO - ModulatedDeformConvPack bbox_head.cls_convs.1.conv is upgraded to version 2.
04/01 12:05:41 - mmengine - INFO - ModulatedDeformConvPack bbox_head.reg_convs.1.conv is upgraded to version 2.
04/01 12:05:42 - mmengine - WARNING - Failed to search registry with scope "mmdet3d" in the "function" registry tree. As a workaround, the current "function" registry in "mmengine" is used to build instance. This may cause unexpected failure when running the built modules. Please check whether "mmdet3d" is a correct scope, or whether the registry is initialized.
/home/user1/miniconda3/envs/openmmlab/lib/python3.8/site-packages/mmengine/visualization/visualizer.py:196: UserWarning: Failed to add <class 'mmengine.visualization.vis_backend.LocalVisBackend'>, please provide the `save_dir` argument.
  warnings.warn(f'Failed to add {vis_backend.__class__}, '
/home/user1/miniconda3/envs/openmmlab/lib/python3.8/site-packages/torch/functional.py:504: UserWarning: torch.meshgrid: in an 
upcoming release, it will be required to pass the indexing argument. (Triggered internally at 
../aten/src/ATen/native/TensorShape.cpp:3526.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
/home/user1/miniconda3/envs/openmmlab/lib/python3.8/site-packages/mmengine/visualization/visualizer.py:831: UserWarning: Warning: 
The polygon is out of bounds, the drawn polygon may not be in the image
  warnings.warn(
/home/user1/miniconda3/envs/openmmlab/lib/python3.8/site-packages/mmengine/visualization/visualizer.py:760: UserWarning: Warning: 
The bbox is out of bounds, the drawn bbox may not be in the image
  warnings.warn(
`

and show this 

![Screenshot from 2024-04-01 11-58-24](https://github.com/open-mmlab/mmdetection3d/assets/104978921/36e72918-ea1f-4b09-ae49-00844bf0ef92)

It shows 3 pedestrians and one car that are not exists .

**Comment by turtlemicro** [comment]
I have the same problem. The issue #1024 discussed this problem and there was a solution, but I didn't solve it yet.

**Reply by damavand1** [reply]
Hi

@turtlemicro 

default value of pred_score_thr is 0.3
and base of this #1024  when i changed it to 5 and result is more better

the question is why we should set 5 instead of 0.3 ? and whats the difference ?
